drivers/mge-hid.c

```diff
@@ -300,28 +396,99 @@ static info_lkp_t eaton_abm_status_info[] = {
 	{ 0, NULL, NULL, NULL }
 };
 
+/* Used to process ABM flag, for battery.charger.type */
+static const char *eaton_abm_charger_type_fun(double value)
+{
+	int abm_charger_type = value;
+
+	if (abm_charger_type == ABM_ENABLED_TYPE)
+	{
+		/* Set ABM flag for battery.charger.type */
+		advanced_battery_monitoring = ABM_ENABLED_TYPE;
+		upsdebugx(2, "ABM charger type: %i", (int)value);
+
+		return "ABM";
+	}
+
+    /* Handle the case when value is not equal to ABM_ENABLED_TYPE_MODE or some other appropriate action */
+	return NULL;
+}
+
+static info_lkp_t eaton_charger_type_info[] = {
+	{ 0, "None", NULL, NULL },
+	{ 1, "Extended (CLA)", NULL, NULL },
+	{ 2, "Large extension", NULL, NULL },
+	{ 3, "Extra large extension (XL)", NULL, NULL },
+	{ 4, "ABM", eaton_abm_charger_type_fun, NULL },
+	{ 5, "Constant Charge (CC)", NULL, NULL },
+	{ 0, NULL, NULL, NULL }
+};
+
 /* Used to process ABM flags, for ups.status (CHRG/DISCHRG/RB) */
 static const char *eaton_abm_chrg_dischrg_fun(double value)
 {
 	/* Don't process if ABM is disabled */
 	if (advanced_battery_monitoring == ABM_DISABLED)
 		return NULL;
```

Comment:
Perhaps we should also handle here a transition from previous ABM enabled to ABM unknown/disabled?
Similar to as is done in the non-ABM functions `eaton_abm_check_chrg_fun`/`eaton_abm_check_dischrg_fun`:
https://github.com/networkupstools/nut/blob/4b1e9258b892cb0fa9b01c80bb12f147d04bf1ee/drivers/mge-hid.c#L449-L453
https://github.com/networkupstools/nut/blob/4b1e9258b892cb0fa9b01c80bb12f147d04bf1ee/drivers/mge-hid.c#L423-L427
Basically cleaning up any previously published ABM flags so the non-ABM mode can take over?

Reply:
Nevermind, this seems not needed here as `eaton_abm_check_chrg_fun`/`eaton_abm_check_dischrg_fun` will always set non-ABM `chrg` and `dischrg` flags overriding no longer needed ABM-flags, in fact implementing such an addition would likely override such already correctly set flags due to function order.